src/core/qgscoordinatetransformcontext.h

```diff
@@ -0,0 +1,270 @@
+                         -------------------------------
+    begin                : November 2017
+    copyright            : (C) 2017 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSCOORDINATETRANSFORMCONTEXT_H
+#define QGSCOORDINATETRANSFORMCONTEXT_H
+
+#include "qgis_core.h"
+#include "qgis.h"
+#include "qgscoordinatetransformcontext_p.h"
+#include "qgscoordinatetransform.h"
+
+class QgsReadWriteContext;
+
+
+/***************************************************************************
+ * This class is considered CRITICAL and any change MUST be accompanied with
+ * full unit tests in testqgsfeature.cpp.
+ * See details in QEP #17
+ ****************************************************************************/
+
+/**
+ * \class QgsCoordinateTransformContext
+ * \ingroup core
+ * Contains information about the context in which a coordinate transform is executed.
+ *
+ * The context stores various information regarding which coordinate transforms should
+ * be used when transforming points from a source to destination coordinate reference
+ * system.
+ *
+ * The highest priority transforms are those set using addSourceDestinationDatumTransform()
+ * and which the transform has a matching source to destination CRS pair.
+ *
+ * Failing this, if the source CRS has a matching transform specified by
+ * addSourceDatumTransform() then this datum transform will be used. The same logic
+ * applies for destination CRS transforms set using addDestinationDatumTransform().
+ *
+ * \note QgsCoordinateTransformContext objects are thread safe for read and write.
+ *
+ * \note QgsCoordinateTransformContext objects are implicitly shared.
+ *
+ * \since QGIS 3.0
+*/
+
+class CORE_EXPORT QgsCoordinateTransformContext
+{
+  public:
+
+    /**
+     * Constructor for QgsCoordinateTransformContext.
+     */
+    QgsCoordinateTransformContext();
+
+    /**
+     * Copy constructor
+     */
+    QgsCoordinateTransformContext( const QgsCoordinateTransformContext &rhs );
+
+    /**
+     * Assignment operator
+     */
+    QgsCoordinateTransformContext &operator=( const QgsCoordinateTransformContext &rhs ) SIP_SKIP;
+
+
+    /**
+     * Clears all stored transform information from the context.
+     */
+    void clear();
+
+
+#if 0
```

Comment:
Is it planned to use this in the future? I.e. Is there any value in keeping this code?